### PR TITLE
fix: restoreVersion validation for localized required fields

### LINKED ISSUE #15698: restoreVersion fails validation on localized and required fields when non-default locales are empty or filled
### Describe the Bug

Restoring a previously published version of a document fails with a validation error on localized:true and required: true fields. This happens in all scenarios — whether all locales are fully filled, partially filled, or the document was published in a single locale only.

### Link to the code that reproduces this issue

https://github.com/StYaroslav/payload-multiple-locales-versioning-error

### Reproduction Steps

1. Clone the repo and setup database (was using mongoDB for test)
2. Create a collection with versions: { drafts: true } and a localized required field
3. Create a document and fill in the title for all locales (en, fr, de) — or even just the default locale
4. Publish the document
5. Make any edit and publish again (so a previous version exists)
6. Navigate to Versions → select the previous version → click "Restore this version"
7. Observe: validation error on the localized required field(s)

I have created fully blank payload project, added localization and one test collections with required and localized fields called Articles

```typescript
// Localization setup in payload-config.ts
localization: {
    locales: ['en', 'fr',  'de'],
    defaultLocale: 'en',
    fallback: false,
  },
``` 

```typescript
// Articles.ts collection
export const Articles: CollectionConfig = {
  slug: 'articles',
  admin: {
    useAsTitle: 'title',
  },
  versions: {
    drafts: true,
    maxPerDoc: 10,
  },
  fields: [
    {
      name: 'title',
      type: 'text',
      required: true,
      localized: true,
    },
    {
      name: 'summary',
      type: 'textarea',
      required: true,
      localized: true,
    },
    {
      name: 'slug',
      type: 'text',
      required: true,
      unique: true,
    },
  ],
}
``` 

Expected Behavior:
The version should be restored successfully, especially when all locales were filled at the time the version was created.

There is also a workaround for such scenario - to use 'restore as draft' option which will omit all validations, but still the issue with main functionality through 'Restore this version' exists.

Single language publish:

https://github.com/user-attachments/assets/52c1c704-861e-42b1-a590-87ccfa371968

All languages published restoration try:

https://github.com/user-attachments/assets/120029c6-cd99-4f8d-8ee1-dfb3574cc5b1

### Which area(s) are affected?

area: core

### Environment Info

```text
Binaries:
  Node: 22.13.1
  npm: 11.1.0
  Yarn: N/A
  pnpm: 10.15.1
Relevant Packages:
  payload: 3.77.0
  next: 15.4.11
  @payloadcms/db-mongodb: 3.77.0
  @payloadcms/graphql: 3.77.0
  @payloadcms/next/utilities: 3.77.0
  @payloadcms/richtext-lexical: 3.77.0
  @payloadcms/translations: 3.77.0
  @payloadcms/ui/shared: 3.77.0
  react: 19.2.1
  react-dom: 19.2.1
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 25.2.0: Tue Nov 18 21:09:34 PST 2025; root:xnu-12377.61.12~1/RELEASE_ARM64_T8112
  Available memory (MB): 16384
  Available CPU cores: 8
```

### PR DESCRIPTION
**Pull request by GermanJablo**
## Summary
- fix collection `restoreVersion` to read version/current docs with all locales before restore processing
- validate restore data in the default locale context while preserving full locale data for persistence
- add an integration regression test for restoring versions when required localized fields are empty in a non-default locale

Closes #15698

## Related
payloadcms/payload#15698

## Changes
- update `packages/payload/src/collections/operations/restoreVersion.ts`
- add coverage in `test/versions/int.spec.ts`

## Testing
- [x] `PAYLOAD_DATABASE=sqlite NODE_OPTIONS="--no-deprecation --no-experimental-strip-types" NODE_NO_WARNINGS=1 DISABLE_LOGGING=true pnpm exec vitest --project int test/versions/int.spec.ts -t "should restore published version with correct data|should restore a published version when required localized fields are empty in a non-default locale"`

### CODE REVIEW

File: packages/payload/src/collections/operations/restoreVersion.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export const restoreVersionOperation = async <
     const { docs: versionDocs } = await req.payload.db.findVersions({
       collection: collectionConfig.slug,
       limit: 1,
-      locale: locale!,
+      locale: 'all',
```
Comment (GermanJablo) [inline packages/payload/src/collections/operations/restoreVersion.ts:83]: restore logic needs full localized data, not just the caller locale.
Without this, localized required fields can appear missing simply because the active locale has no value.

File: packages/payload/src/collections/operations/restoreVersion.ts [hunk 2]
```diff
@@ -143,6 +143,8 @@ export const restoreVersionOperation = async <
     })
 
     // originalDoc with hoisted localized data
+    const validationLocale = payload.config.localization?.defaultLocale || locale
```
Comment (GermanJablo) [inline packages/payload/src/collections/operations/restoreVersion.ts]: field validation/hook flow expects locale-hoisted values and should validate restore against the canonical locale context, not a potentially empty non-default locale.
This is what prevents false validation failures when de is empty but en is valid.

**Comment by JessRynkar** [comment]
This all looks good. Hopefully the `locale` toggles on the version view are not misleading, users should be aware that this function will restore the entire version **across locales**. In the future, we might want to allow this on a per-locale basis.

**Review by JessRynkar — APPROVED** [review]
